### PR TITLE
Support for `purge_soft_delete_on_destroy` in `azurerm_api_management`

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/11061

--- PASS: TestAccApiManagement_purgeSoftDelete (743.86s)

### CODE REVIEW

**Comment by yupwei68** [comment]
@katbyte I've updated to solve the conflicts. Please continue reviewing.

**Review by katbyte — CHANGES_REQUESTED** [review]
thanks @yupwei68 - could we update the docs with this new property?

File: website/docs/index.html.markdown [hunk 1]
```diff
@@ -185,6 +187,12 @@ The `features` block supports the following:
 
 ---
 
+The `api_management` block supports the following:
+
+* `purge_soft_delete_on_destroy` - (Optional) Should the `azurerm_api_management` resources be permanently deleted (e.g. purged) when destroyed? Defaults to `false`.
```
Comment (wasfree) [inline website/docs/index.html.markdown:192]: We should also place a note in [azurerm_api_management](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/api_management) resource referring to this feature flag.

**Review by katbyte — APPROVED** [review]
THANKS @yupwei68 - LGTM 🚀

**Comment by akhileshnirapure** [comment]
The **feature** is missing any explicit action required?

https://github.com/hashicorp/cdktf-provider-azurerm/issues/448

**Pull request merged**